From 02e7d7caaa649333e81c1bb22fc5755a9edbd931 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:06:36 -0400
Subject: [usb] uas: Verify endpoint descriptors from uas_use_uas_driver()

Message-id: <1410887463-20674-325-git-send-email-dzickus@redhat.com>
Patchwork-id: 95290
O-Subject: [RHEL7 PATCH 324/591] uas: Verify endpoint descriptors from uas_use_uas_driver()
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 6134041bef0aeb9cb7c8a8daf045b44513cd8396
Author: Hans de Goede <hdegoede@redhat.com>
Date:   Sat Nov 16 11:37:41 2013 +0100

    uas: Verify endpoint descriptors from uas_use_uas_driver()

    Signed-off-by: Hans de Goede <hdegoede@redhat.com>
    Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/storage/uas-detect.h b/drivers/usb/storage/uas-detect.h
index 8de030a..b8a02e1 100644
--- a/drivers/usb/storage/uas-detect.h
+++ b/drivers/usb/storage/uas-detect.h
@@ -70,13 +70,23 @@ static int uas_find_endpoints(struct usb_host_interface *alt,
 static int uas_use_uas_driver(struct usb_interface *intf,
          const struct usb_device_id *id)
 {
+ struct usb_host_endpoint *eps[4] = { };
  struct usb_device *udev = interface_to_usbdev(intf);
  unsigned long flags = id->driver_info;
+ int r, alt;
 
  usb_stor_adjust_quirks(udev, &flags);
 
  if (flags & US_FL_IGNORE_UAS)
   return 0;
 
- return uas_find_uas_alt_setting(intf) >= 0;
+ alt = uas_find_uas_alt_setting(intf);
+ if (alt < 0)
+  return 0;
+
+ r = uas_find_endpoints(&intf->altsetting[alt], eps);
+ if (r < 0)
+  return 0;
+
+ return 1;
 }
-- 
1.7.1